osu.Game.Rulesets.Mania/Mods/ManiaModConstant.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Mods;
+
+namespace osu.Game.Rulesets.Mania.Mods
+{
+    public class ManiaModConstant : Mod, IApplicableToBeatmap
+    {
+        public override string Name => "Constant";
+        public override string Acronym => "CT";
+        public override double ScoreMultiplier => 1;
+        public override string Description => "No more tricky note speed changes!";
+        public override IconUsage? Icon => FontAwesome.Solid.ArrowDown;
+        public override ModType Type => ModType.Conversion;
+        public override bool Ranked => false;
+
+        public void ApplyToBeatmap(IBeatmap beatmap)
+        {
+            // Get all difficulty points
+            var difficultyPoints = beatmap.ControlPointInfo.DifficultyPoints;
+
+            foreach (var difficultyPoint in difficultyPoints)
+            {
+                // Set this difficulty point's speed multiplier to 1
+                difficultyPoint.SpeedMultiplier = 1;
```

Comment:
So what's the plan regarding BPM changes here? I don't think generating a correct set of difficulty control points that would compensate BPM would be overly difficult, so why is this not done? Are you expecting turning off BPM scaling to become a game-wide feature?

Reply:
The compensation for BPM didn't cross my mind, I'll look into that. I do not expect BPM scaling to change. The main issue I was facing was that if I were to reset all timing points as well, there would be the chance that an actual change in BPM on the map would be reset and make the rest of the map off-sync.

Reply:
I think adding a difficulty control point for each timing control point, with multiplier equal to (initial BPM / current BPM) should do the trick. In fact I think that is already how constant scrolling speed is mapped? And yeah, just nuking timing points isn't going to work.